src/main/java/haste/data/TaskList.java

```diff
@@ -0,0 +1,78 @@
+package haste.data;
+
+import haste.tasks.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of Task objects.
+ */
+public class TaskList {
+    public ArrayList<Task> taskList;
```

Comment:
Class variables should never be declared public: Here, taskList already has all the functions necessary that both taskList and numOfTasks can be made private w/o any other changes.